packages/squiggle-lang/src/rescript/FunctionRegistry/FunctionRegistry_Core.res

```diff
@@ -0,0 +1,304 @@
+type expressionValue = ReducerInterface_ExpressionValue.expressionValue
+
+/*
+  Function Registry "Type". A type, without any other information.
+  Like, #Float
+*/
+type rec frType =
+  | FRTypeNumber
+  | FRTypeNumeric
+  | FRTypeDistOrNumber
+  | FRTypeRecord(frTypeRecord)
+  | FRTypeArray(array<frType>)
+  | FRTypeOption(frType)
+and frTypeRecord = array<frTypeRecordParam>
+and frTypeRecordParam = (string, frType)
+
+/*
+  Function Registry "Value". A type, with the information of that type.
+  Like, #Float(40.0)
+*/
+type rec frValue =
+  | FRValueNumber(float)
+  | FRValueDist(DistributionTypes.genericDist)
+  | FRValueOption(option<frValue>)
+  | FRValueDistOrNumber(frValueDistOrNumber)
+  | FRValueRecord(frValueRecord)
+and frValueRecord = array<frValueRecordParam>
+and frValueRecordParam = (string, frValue)
+and frValueDistOrNumber = FRValueNumber(float) | FRValueDist(DistributionTypes.genericDist)
+
+type fnDefinition = {
```

Comment:
I think I like the idea of renaming this to fnSignature, similar to @umuro's PR: https://github.com/quantified-uncertainty/squiggle/pull/546/files